airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/StatusView.tsx

```diff
@@ -145,6 +170,11 @@ const StatusView: React.FC<StatusViewProps> = ({ connection }) => {
       >
         {jobs.length ? <JobsList jobs={jobs} /> : <EmptyResource text={<FormattedMessage id="sources.noSync" />} />}
       </ContentCard>
+      <footer className={styles.footer}>
+        <Button onClick={onLoadMoreJobs} disabled={jobs.length < jobPageSize}>
```

Comment:
Two things:

1) I think it would be cool if we could show the button in a loading state while we're still loading the query. To do that we must use `useQuery` directly instead of our custom `useSuspenseQuery` wrapper in the `JobService` so we can get access to the underlying react-query properties. There is ` isPreviousValue` which will be true while we're still loading the next data but show the previous, which we can use as an indicator that we're still loading.

2) In general I'd consider it good UX to not hide buttons but disable them if they are not available. For this specific case of paginating an "infinite list" I feel it might be more appropriate (and more commonly used as a pattern) to actually hide the button and not disable it, if there are no further items to paginate over. Since a disabled button might leave the question: "Why can't I load more jobs anymore?" and looks more like an issue than "the end of the list".

Reply:
@timroes thanks for the suggestions, both make sense to me!

I have pushed up the changes to accomplish this, and I think it looks cleaner now. 

I recorded a new video to demonstrate how it looks now, and to also show one small quirk of this implementation (I lowered the page size increment from 25 to 2 for this demonstration to make it a bit easier to show the behavior, but I've kept it as 25 in the PR):
https://www.loom.com/share/edc11614687b4f4f8b60f348abb982d4

Notice that at the end of the video, I click the `Load more` button, which causes it to load, but then it disappears. This is because the total number of jobs for this connection is a multiple of the page size increment (2 in that demonstration). So when all jobs are loaded, the logic of `moreJobPagesAvailable = jobs && jobs.length === jobPageSize` evaluates to true, because our code can only make an assumption about there being more jobs available based on the fact that we displaying the full page size number of jobs. 

It isn't until we click the `Load more` button _once more_ that our code can understand that the page size is higher than the number of jobs being returned by the API, so we can assume there are no more jobs to load.

In production, I doubt anyone will run into this case very often, as it would require that the number of jobs for their connection is an exact multiple of 25. And regardless, I don't think the UX for this is all that confusing, since the user clicks `Load more`, it loads, and the button disappears, indicating that there are no more jobs to show. 

Sorry if what I said above was super verbose or confusing - I just wanted to confirm that that edge case is acceptable behavior!

Reply:
(also, the style of the loading state of the button in that loom is a bit different than in the current state of the PR, since I just now realized we have a `LoadingButton` component. I have switched to using that instead of inserting a spinner into the button conditionally)

Reply:
> In production, I doubt anyone will run into this case very often, as it would require that the number of jobs for their connection is an exact multiple of 25. And regardless, I don't think the UX for this is all that confusing, since the user clicks Load more, it loads, and the button disappears, indicating that there are no more jobs to show.

I thought about that as well when I suggested that and feel it's an "okay" behavior int his case. Given we'll still do the request we're shortly in a loading state and then hide the button, which feels like users get the idea that there is nothing more to add. If we'd want to solve that really well, we'd need to adjust the API to also return us the `totalCount` as part of the response (which shouldn't be too expensive to get via SQL directly), so we can basically use this as a boundary.

Reply:
Yep exactly. I think this solution is good enough for now, but if we see complaints about this we can do the `totalCount` solution at that time